.github/workflows/docker.yml

```diff
@@ -0,0 +1,54 @@
+name: Docker
+
+on:
+  push:
+    branches:
+      - 'main'
+  # TODO: remove before merging to main
+  pull_request:
+
+env:
+  IMAGE_NAME: backend
+
+jobs:
+  docker:
+    name: Docker
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      # Only build and push docker image if this action is triggered from the main branch.
+      - name: Build Docker image
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: docker build . --file Dockerfile --tag $IMAGE_NAME --label "runnumber=${GITHUB_RUN_ID}"
+
+      - name: Log in to Github Package registry
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: echo "${{ secrets.GITHUB_TOKEN}}" | docker login ghcr.io -u $ --password-stdin
+
+      - name: Push image to registry
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: |
+          IMAGE_ID=ghcr.io/${{ github.repository_owner }}/$IMAGE_NAME
+
+          # Change all uppercase to lowercase
+          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          # Strip "v" prefix from tag name
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "main" ] && VERSION=latest
+          echo IMAGE_ID=$IMAGE_ID
+          echo VERSION=$VERSION
+          docker tag $IMAGE_NAME $IMAGE_ID:$VERSION
+          docker push $IMAGE_ID:$VERSION
+
+      - name: Trigger event to execute integration tests
+        uses: peter-evans/repository-dispatch@v2
+        if: ${{ github.ref == 'refs/heads/main' }}
+        with:
+          token: ${{ secrets.REPO_ACCESS_TOKEN }}
```

Comment:
Who will be adding this PAT to this repo so the action can use it? I believe the action needs this PAT to be created by someone who has `write` access to the target repo specified below.

Reply:
@aomi

Reply:
This should just be a token to access this repo, Sam and I definitely have that permission. I can create a PAT if none exist already